### PR TITLE
Feature/drops

### PR DESCRIPTION
**Pull request by alexanderattar**
## Describe Changes

- Add Foundry toolchain and start initial test setup
- Start integrating drops contracts
- Migrating from original drop contract deployment to use Holograph protocol
- Flattening test logic to pass values directly to the edition function
- Add deployment config and signature params
- Flattens inherited contracts into the local contracts directory so that they can be modified / customized to support "holographable" versions of the drops / editions
- Update the tests to be compatible
- Add some new tests for holographable contracts

## Checklist before requesting a review

- [x] I have performed a self-review of my code
- [x] Code styles have been enforced
- [x] All Hardhat tests are passing

### CODE REVIEW

File: src/drops/ERC721Drop.sol [hunk 1]
```diff
@@ -57,30 +60,30 @@ contract ERC721Drop is
   ERC721DropStorageV1
 {
   /// @dev This is the max mint batch size for the optimized ERC721A mint contract
-  uint256 internal immutable MAX_MINT_BATCH_SIZE = 8;
+  uint256 constant MAX_MINT_BATCH_SIZE = 8;
```
Comment (alexanderattar) [inline src/drops/ERC721Drop.sol]: what's the reason for changing `internal immutable` to `constant`? constructor vs initialize?

File: src/drops/ERC721Drop.sol [hunk 2]
```diff
@@ -57,30 +60,30 @@ contract ERC721Drop is
...
 
   /// @dev Holograph Fee Manager address
-  IHolographFeeManager public immutable holographFeeManager;
+  IHolographFeeManager public holographFeeManager;
 
   /// @notice Max royalty BPS
   uint16 constant MAX_ROYALTY_BPS = 50_00;
 
-  address immutable marketFilterDAOAddress;
+  addresspublic marketFilterDAOAddress;
```
Comment (alexanderattar) [inline src/drops/ERC721Drop.sol]: typo?

File: src/drops/HolographNFTCreatorProxy.sol [hunk 3]
```diff
@@ -1,9 +1,61 @@
...
-pragma solidity ^0.8.13;
+/*HOLOGRAPH_LICENSE_HEADER*/
 
-import {ERC1967Proxy} from "@openzeppelin/contracts/proxy/ERC1967/ERC1967Proxy.sol";
+/*SOLIDITY_COMPILER_VERSION*/
 
-/// @dev Holograph NFT Creator Proxy Access Contract
-contract HolographNFTCreatorProxy is ERC1967Proxy {
-  constructor(address _logic, bytes memory _data) payable ERC1967Proxy(_logic, _data) {}
+import "../abstract/Admin.sol";
```
Comment (alexanderattar) [inline src/drops/HolographNFTCreatorProxy.sol]: what's is `src/drops/HolographNFTCreatorProxy.sol` vs `src/drops/ERC721DropProxy.sol`

Comment (alexanderattar) [reply src/drops/HolographNFTCreatorProxy.sol]: oh yeah i see this is for the `src/drops/HolographNFTCreatorV1.sol` instead

File: src/drops/ERC721Drop.sol [hunk 4]
```diff
@@ -66,8 +66,8 @@ contract ERC721Drop is
   uint256 constant FUNDS_SEND_GAS_LIMIT = 210_000;
 
   /// @notice Access control roles
-  bytes32 constant MINTER_ROLE = keccak256("MINTER");
-  bytes32 constant SALES_MANAGER_ROLE = keccak256("SALES_MANAGER");
+  bytes32 public MINTER_ROLE = keccak256("MINTER");
```
Comment (alexanderattar) [inline src/drops/ERC721Drop.sol]: why add public and remove constant on these?

File: src/drops/ERC721Drop.sol [hunk 5]
```diff
@@ -81,9 +81,9 @@ contract ERC721Drop is
   /// @notice Max royalty BPS
   uint16 constant MAX_ROYALTY_BPS = 50_00;
 
-  addresspublic marketFilterDAOAddress;
+  address public marketFilterDAOAddress;
```
Comment (alexanderattar) [inline src/drops/ERC721Drop.sol]: looks like typo was fixed 👍

File: src/drops/ERC721DropProxy.sol [hunk 6]
```diff
@@ -7,7 +7,7 @@ import "../abstract/Initializable.sol";
 
 import "../interface/InitializableInterface.sol";
 
-contract HolographErc721DropProxy is Admin, Initializable {
+contract ERC721DropProxy is Admin, Initializable {
```
Comment (alexanderattar) [inline src/drops/ERC721DropProxy.sol]: I like this naming better 👍

File: src/drops/utils/PublicMulticall.sol [hunk 7]
```diff
@@ -7,7 +7,7 @@ abstract contract PublicMulticall {
   /**
    * @dev Receives and executes a batch of function calls on this contract.
    */
-  function multicall(bytes[] calldata data) public virtual returns (bytes[] memory results) {
+  function multicall(bytes[] memory data) public virtual returns (bytes[] memory results) {
```
Comment (alexanderattar) [inline src/drops/utils/PublicMulticall.sol]: why needed to be updated from calldata to memory? data needs to be manipulated now?

File: test/forge/HolographNFTCreatorV1.t.sol [hunk 8]
```diff
@@ -24,13 +24,29 @@ contract HolographNFTCreatorV1Test is Test {
...
+        new bytes(0)
+      )
+    );
     editionMetadataRenderer = new EditionMetadataRenderer();
     dropMetadataRenderer = new DropMetadataRenderer();
     // do not use constructor arguments
     HolographNFTCreatorV1 impl = new HolographNFTCreatorV1();
     // init source deployment with null/zero values
-    impl.init(abi.encode(address(0), EditionMetadataRenderer(address(0)), DropMetadataRenderer(address(0))));
+    impl.init(abi.encode(address(0x0), EditionMetadataRenderer(address(0x0)), DropMetadataRenderer(address(0x0))));
```
Comment (alexanderattar) [inline test/forge/HolographNFTCreatorV1.t.sol]: any particular reason for updating these to use `0x`?

**Review by alexanderattar — COMMENTED** [review]
Left some questions that would be cool to chat about

**Review by ACC01ADE — APPROVED** [review]

**Review by sogoiii — APPROVED** [review]
👍

**Review by ACC01ADE — APPROVED** [review]

**Pull request merged**